### DISCUSSION TITLE
Timeout while sending an email

### DISCUSSION THREAD

**Discussion by karimennassiri** [plausible/analytics#1537] [Self-Hosted Support]
I installed Plausible in my cluster and everything works fine except sending mail. At the same time, I installed an smtp service that I configured. When I use it to send emails from plausible I get the following error message

```
** (exit) an exception was raised:
    ** (ArgumentError) raise/1 and reraise/2 expect a module name, string or exception as the first argument, got: {:network_failure, 'smtp', {:error, :timeout}}
        (bamboo 2.2.0) lib/bamboo/mailer.ex:234: Bamboo.Mailer.deliver_now!/4
        (plausible 0.0.1) lib/plausible/mailer.ex:6: Plausible.Mailer.send_email/1
        (plausible 0.0.1) lib/plausible_web/controllers/site/membership_controller.ex:59: PlausibleWeb.Site.MembershipController.invite_member/2
        (plausible 0.0.1) lib/plausible_web/controllers/site/membership_controller.ex:1: PlausibleWeb.Site.MembershipController.action/2
        (plausible 0.0.1) lib/plausible_web/controllers/site/membership_controller.ex:1: PlausibleWeb.Site.MembershipController.phoenix_controller_pipeline/2
        (phoenix 1.5.13) lib/phoenix/router.ex:352: Phoenix.Router.__call__/2
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint.plug_builder_call/2
        (plausible 0.0.1) lib/plausible_web/endpoint.ex:1: PlausibleWeb.Endpoint."call (overridable 3)"/2
```

I thought I misconfigured my service, however I also tried to use the gmail service and got the same error.

```
** (exit) an exception was raised:
    ** (ArgumentError) raise/1 and reraise/2 expect a module name, string or exception as the first argument, got: {:network_failure, 'smtp-relay.gmail.com', {:error, :timeout}}

```

I'm using Traefik as revese proxy and maybe it is the main reason of this problem

**Comment by karimennassiri** [comment]
Any help for this please ? still blocked :)

**Reply by FathiQ** [reply]
Make sure to add this option CRON_ENABLED=true, change SMTP_HOST_ADDR env variable, and point it directly to your SMTP host. worked for me.

**Comment by jordymeow** [comment]
Hello @karimennassiri, @FathiQ,

I’m running into the exact same issue. Even on the latest Plausible versions, debugging is extremely difficult. I’m using SMTP_HOST_ADDR as documented, and I also tried enabling CRON_ENABLED just in case, but nothing changes. Plausible always crashes with the same message:

`raise/1 and reraise/2 expect a module name, string or exception as the first argument`

The logs don’t explain what’s actually failing on the SMTP side, so it’s very hard to diagnose. If you’ve found a reliable workaround or additional logging options, I’d appreciate any pointers.
